### PR TITLE
Dream 1.0.0~alpha2: tidy Web framework

### PR DESCRIPTION
**Pull request by aantron**
- [Project page](https://github.com/aantron/dream)
- [API](https://aantron.github.io/dream)

[Changelog](https://github.com/aantron/dream/releases/tag/1.0.0-alpha2):

> Changes
>
> - Rename `Dream.from_target` to [`Dream.split_target`](https://aantron.github.io/dream/#val-split_target) (aantron/dream@a3c1508).
> - Rename `Dream.from_target_path` to [`Dream.from_path`](https://aantron.github.io/dream/#val-from_path) (aantron/dream@a3c1508).
> - [`Dream.redirect`](https://aantron.github.io/dream/#val-redirect) now needs a `request` as argument (aantron/dream@90bba6e).
> - Reorder arguments of [`Dream.write`](https://aantron.github.io/dream/#val-write), [`Dream.write_bigstring`](https://aantron.github.io/dream/#val-write_buffer), [`Dream.verify_csrf_token`](https://aantron.github.io/dream/#val-verify_csrf_token), [`Dream.send`](https://aantron.github.io/dream/#val-send), [`Dream.sql`](https://aantron.github.io/dream/#val-sql) (aantron/dream@c1ba95a).
> - Improve how dependencies are vendored (aantron/dream@343e912, suggested by Antonio Nuno Monteiro).
> - Rename `Dream.identity` to [`Dream.no_middleware`](https://aantron.github.io/dream/#val-no_middleware) (aantron/dream@42625d1).
> - Use library [`multipart_form`](https://github.com/dinosaure/multipart_form) instead of `multipart-form-data` (aantron/dream#45, Calascibetta Romain).
> - Rename `Dream.bigstring` to [`Dream.buffer`](https://aantron.github.io/dream/#type-buffer) (aantron/dream@1386bf3).
>
> Additions
>
> - Example [**`r-graphql`**](https://github.com/aantron/dream/tree/a0dcaf5b4729b24a37c89001bd23343e47190979/example/r-graphql#files) (aantron/dream#51, Tom Ekander).
> - Example [**`w-live-reload`**](https://github.com/aantron/dream/tree/a0dcaf5b4729b24a37c89001bd23343e47190979/example/w-live-reload#files) (aantron/dream#52, Thibaut Mattio).
> - Better error template (aantron/dream#53, Thibaut Mattio).
> - Multi-stage Docker deployment build and opam instructions (aantron/dream#57, Thibaut Mattio).
> - [`Dream.to_path`](https://aantron.github.io/dream/#val-to_path) (aantron/dream@a3c1508).
> - Expose [`Dream.from_filesystem`](https://aantron.github.io/dream/#val-from_filesystem) (aantron/dream#14).
> - [`Dream.mime_lookup`](https://aantron.github.io/dream/#val-mime_lookup) (aantron/dream#14).
> - [`Dream.no_route`](https://aantron.github.io/dream/#val-no_route) (aantron/dream@37a27c4).
> - [Playground](http://dream.as).
> - [Deployment instructions](https://github.com/aantron/dream/tree/a0dcaf5b4729b24a37c89001bd23343e47190979/example#deploying).
>
> Fixes
>
> - Work around paurkedal/ocaml-caqti#68 (aantron/dream@385fa45, reported by Konstantin Olkhovskiy).
> - Require Cstruct 6.0.0 (aantron/dream#60, reported by Edwin Török).

### CODE REVIEW

**Comment by aantron** [comment]
All the CI failures seem to be caused by upstream projects.

File: packages/dream/dream.1.0.0~alpha1/opam [hunk 1]
```diff
@@ -58,8 +58,9 @@ depends: [
   "graphql-lwt"
   "hmap"
   "lwt"
-  "lwt_ppx"
+  "lwt_ppx" {>= "1.2.2"}
```
Comment (aantron) [inline packages/dream/dream.1.0.0~alpha1/opam:61]: Before I cherry-pick this into Dream, I want to ask: I'm under the impression that Dream should build even with `lwt_ppx` < 1.2.2. Should it deliberately exclude earlier versions in order to enforce Dune? Did it fail to build?

Comment (kit-ty-kate) [reply packages/dream/dream.1.0.0~alpha1/opam:61]: full log: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/000177c2a47359c71a41e9fc78d084954efdf708/variant/opam-2.1,compilers,4.08,dream.1.0.0~alpha2,lower-bounds

**Comment by kit-ty-kate** [comment]
Thanks!

**Comment by aantron** [comment]
Thanks for the review and the constraints work! I'll upstream them now.

**Pull request merged**